### PR TITLE
fix UEFI lib

### LINKED ISSUE #542: broken src/lib/uefi parsing and tests
1. The crate `src/lib/uefi` broke in `5beeb5641124c240a06bb889605e2f44485cbf26`, where the loops were removed that iterate over the FV (firmware volume).
2. Its tests broke in `b8f0e5100abbdbd396ac884e827bce5e2cabb4c0` where the `fsp` submodule was updated, but the test cases depend on the binaries in it (12 of them).

The first issue is easy to fix, but the test setup is fragile.

Question: What should we prefer, a second fsp submodule side by side with the uefi lib, pinned to its version for testing, or checking the respective files (some MB involved) into the repo (maybe reduce them? 12 feels like it's not really providing value), or... any other ideas?

### PR DESCRIPTION
**Pull request by orangecms**
this resolves #542

### CODE REVIEW

File: src/lib/uefi/src/lib.rs [hunk 1]
```diff
@@ -209,11 +209,10 @@ where
         }
 
         // Iterate through files.
-        let res = {
+        while {
```
Comment (gogo2464) [inline src/lib/uefi/src/lib.rs:212]: Why do you change this. Did this use to work?

**Comment by gogo2464** [comment]
Sounds fine. It does not break the tests and the fsp-bin name sounds better.

File: src/lib/uefi/src/lib.rs (hunk 1 above)
Comment (gogo2464) [reply src/lib/uefi/src/lib.rs:212]: It probably use to break tests. I remember.

**Review by gogo2464 — APPROVED** [review]

File: src/lib/uefi/src/lib.rs (hunk 1 above)
Comment (orangecms) [reply src/lib/uefi/src/lib.rs:212]: Yes, it's restored from the original implementation, see the commit message.

**Pull request merged**